### ISSUE TITLE
Tree and tree item - unable to navigate to after making a selection by the keyboard

### ISSUE THREAD

**Issue by yohannlouis** [Esri/calcite-design-system#7898]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Summary

When using the keyboard allow the user to navigate to the tree after selecting a tree item.

### Actual Behavior

When using the keyboard to navigate and select the tree item, it collapses the tree and it can be no longer navigated to.

https://github.com/Esri/calcite-design-system/assets/142255293/e7615574-a4ac-43b7-801d-0367312ff78a

### Expected Behavior

When using the keyboard allow the user to navigate to the tree after selecting a tree item.

### Reproduction Sample

documentation sample https://developers.arcgis.com/calcite-design-system/components/tree/

### Reproduction Steps

1. Open the calcite-design-system > components > tree sample https://developers.arcgis.com/calcite-design-system/components/tree/
2. Using the keyboard, navigate to a tree item and make a selection by hitting enter.
3. Notice the tree collapses
4. Try navigating to the tree or tree item. Notice it can no longer be navigated to

### Reproduction Version

1.8.0

### Regression?

Last working version [`1.3.0`](https://codepen.io/geospatialem/pen/YzdvYrJ)

### Priority impact

p3 - want for upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Online

**Comment by jcfranco** [comment]
[Belated] Dug into this a bit and I believe we should be able to tackle this one for the current release. Picking it up.

**Comment by yohannlouis** [comment]
@jcfranco where can I verify these changes? Unable to see them in Devext or https://developers.arcgis.com/calcite-design-system/components/tree/

**Comment by geospatialem** [comment]
> @jcfranco where can I verify these changes? Unable to see them in Devext or the sample in https://developers.arcgis.com/calcite-design-system/components/tree/

@yohannlouis Great question, put together a [Codepen](https://codepen.io/geospatialem/pen/BavOpVm) for testing, which used our latest release of `1.9.0-next.16` as of 10/2/2023 at 12:00pm PDT.

![verify-7898](https://github.com/Esri/calcite-design-system/assets/5023024/52a345c8-2870-4517-88d5-eb50f48f460d)
